mypy/typeanal.py

```diff
@@ -552,10 +565,13 @@ def anal_var_defs(self, var_defs: List[TypeVarDef]) -> List[TypeVarDef]:
                                 vd.line))
         return a
 
-    def named_type(self, fully_qualified_name: str, args: List[Type] = None) -> Instance:
+    def named_type(self, fully_qualified_name: str,
+                   args: List[Type] = None,
+                   line: int = -1,
+                   column: int = -1) -> Instance:
         node = self.lookup_fqn_func(fully_qualified_name)
         assert isinstance(node.node, TypeInfo)
-        return Instance(node.node, args or [])
+        return Instance(node.node, args or [], line=line, column=column)
```

Comment:
Maybe while you at it you can also fix type arguments here too?
```python
args or [AnyType()] * len(node.node.defn.type_vars)
```
So that we will never create instances with wrong number of type arguments (like with ``tuple`` above on line 180)
This could lead to nasty crashes like #3117